CircleSpace/CircleSpaceServiceLib/Service/SqlCricleSpaceService.cs

```diff
@@ -0,0 +1,344 @@
+                pages.ForEach(page =>
+                {
+                    PageModel p = PageToPageModel(page);
+                    list.Add(p);
+                });
+            }
+
+            return list;
+        }
+
+        public List<LayoutModel> GetLayouts()
+        {
+            List<LayoutModel> list = new List<LayoutModel>();
+            using (var db = new CircleSpaceEntities())
+            {
+                var query = db.Layouts.Select(x => x);
+                list = LayoutsToListOfLayouts(query.ToList());
+            }
+
+            return list;
+        }
+
+        public LayoutModel GetLayoutWithID(int id)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.ID == id).First());
+            }
+            return l;
+        }
+
+        public LayoutModel GetLayoutWithOwner(UserModel model)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.OwnerID == model.ID).First());
+            }
+            return l;
+        }
+
+        public LayoutModel GetLayoutWithTag(params string[] tags)
+        {
+            //LayoutModel l = new LayoutModel();
+            //using (var db = new CircleSpaceEntities())
+            //{
+            throw new NotImplementedException();
+
+            //}
+        }
+
+        public LayoutModel GetLayoutWithType(LayoutTypes type)
+        {
+            LayoutModel l = new LayoutModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                l = LayoutToLayoutModel(db.Layouts.Where(x => x.LayoutType == type.ToString()).First());
+            }
+            return l;
+        }
+
+        public PageModel GetPageRoute(string route)
+        {
+            PageModel p = new PageModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                p = PageToPageModel(db.Pages.Where(x => x.PageRoute == route).First());
+            }
+            return p;
+        }
+
+        public PageModel GetPageWithID(int id)
+        {
+            PageModel p = new PageModel();
+            using (var db = new CircleSpaceEntities())
+            {
+                p = PageToPageModel(db.Pages.Where(x => x.ID == id).First());
+            }
+            return p;
+        }
+
+        public void UpdateLayout(LayoutModel model)
+        {
+            using (var db = new CircleSpaceEntities())
+            {
+                var layout = db.Layouts.Where(x => x.ID == model.ID).First();
+                layout.LayoutTitle = model.LayoutTitle;
+                layout.LayoutType = model.Type.ToString();
+                layout.Content = model.Content;
+                layout.CSS = model.CSS;
+                model.Tags.ForEach(tag =>
```

Comment:
You're adding new tags instead of updating the existing list. Consider when a tag is removed from a layout. You should delete all tags from the tag table that apply to the layout and then add the new ones.